pkg/safe/safe_test.go

```diff
@@ -23,82 +23,172 @@ import (
 	"testing"
 
 	os2 "github.com/minio/minio/pkg/x/os"
-	. "gopkg.in/check.v1"
 )
 
-func Test(t *testing.T) { TestingT(t) }
-
 type MySuite struct {
 	root string
 }
 
-var _ = Suite(&MySuite{})
-
-func (s *MySuite) SetUpSuite(c *C) {
+func (s *MySuite) SetUpSuite(t *testing.T) {
 	root, err := ioutil.TempDir(os.TempDir(), "safe_test.go.")
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
 	s.root = root
 }
 
-func (s *MySuite) TearDownSuite(c *C) {
-	err := os.Remove(s.root)
-	c.Assert(err, IsNil)
+func (s *MySuite) TearDownSuite(t *testing.T) {
+	err := os.RemoveAll(s.root)
+	if err != nil {
+		t.Fatal(err)
+	}
 }
 
-func (s *MySuite) TestSafeAbort(c *C) {
+func TestSafeAbort(t *testing.T) {
+	s := &MySuite{}
+	s.SetUpSuite(t)
+	defer s.TearDownSuite(t)
+
 	f, err := CreateFile(path.Join(s.root, "testfile-abort"))
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	_, err = os2.Stat(path.Join(s.root, "testfile-abort"))
-	c.Assert(err, Not(IsNil))
+	if !os.IsNotExist(err) {
+		t.Fatal(err)
+	}
 	err = f.Abort()
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
 	err = f.Close()
-	c.Assert(err.Error(), Equals, "close on aborted file")
+	if err != nil {
+		if err.Error() != "close on aborted file" {
+			t.Fatal(err)
+		}
+	}
 }
 
-func (s *MySuite) TestSafeClose(c *C) {
+func TestSafeClose(t *testing.T) {
+	s := &MySuite{}
+	s.SetUpSuite(t)
+	defer s.TearDownSuite(t)
+
 	f, err := CreateFile(path.Join(s.root, "testfile-close"))
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	_, err = os2.Stat(path.Join(s.root, "testfile-close"))
-	c.Assert(err, Not(IsNil))
+	if !os.IsNotExist(err) {
+		t.Fatal(err)
+	}
+
 	err = f.Close()
-	c.Assert(err, IsNil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	_, err = os2.Stat(path.Join(s.root, "testfile-close"))
-	c.Assert(err, IsNil)
+	if err != nil {
```

Comment:
Why are we not using the following check

if !os.IsNotExist(err) {

like in the other places above?

Reply:
In this case we don't expect error, we expect the file be available.. and successful.